py/desisim/scripts/quickquasars.py

```diff
@@ -372,8 +374,39 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
         transmission = transmission[selection]
         metadata = metadata[:][selection]
         DZ_FOG = DZ_FOG[selection]
-
-
+    
+    # Redshift distribution resample to match the one in file give by args.dn_dzdm
+    if eboss is None:
+        log.info("Resampling to redshift distribution from {}".format(args.dn_dzdm))
+        rnd_state = np.random.get_state()
+        hdul_dn_dzdm=pyfits.open(args.dn_dzdm)
+        zcenters=hdul_dn_dzdm['Z_CENTERS'].data
+        dz = 0.5*(zcenters[1]-zcenters[0]) # Get bin size of the distribution
+        dn_dzdm=hdul_dn_dzdm['dn_dzdm'].data
+        dn_dz=dn_dzdm.sum(axis=1) # Table has a redshift bin by row.
+
+        z=metadata['Z']
+        pixarea=healpy.pixelfunc.nside2pixarea(nside,degrees=True) 
+        # Turn old distribution into new distribution
+        selection_z=[] #Initialize array to select qsos
+        for z_bin, dndz_bin in zip(zcenters,dn_dz):
+            nqso_bin=np.ceil(pixarea*dndz_bin).astype(int)
+            w_z = (z>=z_bin-dz)&(z<z_bin+dz)
+            nqso_orig = len(z[w_z])
+            if nqso_orig==0:
+                continue # If no QSOs in that bin, skip
+            idx = np.where(w_z)[0]
+            downsampling_bin = nqso_bin/nqso_orig
+            if downsampling_bin<1:
```

Comment:
We might want to add a warning, or even crash, if the required downsampling is larger than 1.